From 749cd38381d0a6d5c45c913bdebe8523f143078f Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:43 -0400
Subject: [scsi] hpsa: Checking for a NULL return from a kzalloc call

Message-id: <1410186734-19986-53-git-send-email-thenzl@redhat.com>
Patchwork-id: 92079
O-Subject: [RHEL6.6 e-stor PATCH 52/83] hpsa: Checking for a NULL return from a kzalloc call
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Checking for a NULL return from a kzalloc call in hpsa_get_pdisk_of_ioaccel2.

Signed-off-by: Scott Teel <scott.teel@hp.com>
Signed-off-by: Joe Handzik <joseph.t.handzik@hp.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 3b51a7a3910fd5d4165600a64000a9ecf65835f7)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 663ad64..eab81eb 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -2854,6 +2854,8 @@ static int hpsa_get_pdisk_of_ioaccel2(struct ctlr_info *h,
 
  /* Get the list of physical devices */
  physicals = kzalloc(reportsize, GFP_KERNEL);
+ if (physicals == NULL)
+  return 0;
  if (hpsa_scsi_do_report_phys_luns(h, (struct ReportLUNdata *) physicals,
   reportsize, extended)) {
   dev_err(&h->pdev->dev,
-- 
1.7.1